### ISSUE TITLE
Serialization and Deserialization for MultiMessageIdImpl

### ISSUE THREAD

**Issue by jiazhai** [apache/pulsar#4940]
In PR #4911, we added MultiMessageIdImpl, which contains a Map of <topic, MessageId>, that could be used for partition/multi-topics/pattern consumer methods, e.g. seek(), ackCumulative(), getLastMessageId().

In streaming applications, they tends to store messageId along with the state. so that the streaming applications can restore the state back. we need to provide a consistent behavior for how to serialize and deserialize different message id implementations.

This issue is to track and discuss uniforming the serialization and deserialization of message id implementations (basically we need the ability to deserialize the message ids from the byte array in a consistent way).

**Comment by gvolpe** [comment]
Hi folks, has this issue been discussed somewhere? I just came across this limitation when trying out `ackCumulativeAsync` in my application, and I would like to help to get it fixed.

**Comment by tisonkun** [comment]
@BewareMyPower IIRC you're working on the message ID related stuff now. Do you have this issue in mind?

**Comment by BewareMyPower** [comment]
Yeah. I think we should remove the `MultiMessageIdImpl` class and disable `getLastMessageId` API for the multi-topics consumer. Instead, we should add a new API. But I'm still trying to figure out a good way and preparing a proposal.

**Comment by tisonkun** [comment]
@BewareMyPower Make sense. Then you can take over this issue and close it with a fix or superseding proposal :)
